.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,162 @@
+name: Build Pull Request
+on:
+  - pull_request 
+env:
+  ARCH: amd64
+  TAG: v2.8-${{ github.sha }}
+  GIT_TAG: v2.8-${{ github.sha }}
+  HEAD_TAG: v2.8-${{ github.sha }}
+  COMMIT: ${{ github.sha }}
+  REPOSITORY_OWNER: ${{ github.repository_owner }}
+  IMAGE: ${{ github.repository_owner }}/rancher
+  IMAGE_AGENT: ${{ github.repository_owner }}/rancher-agent
+  IMAGE_INSTALLER: ${{ github.repository_owner }}/system-agent-installer-rancher
+  CATTLE_KDM_BRANCH: dev-v2.8
+  HELM_UNITTEST_VERSION: 0.3.2
+  HELM_URL_V3: https://get.helm.sh/helm-v3.13.3-linux-amd64.tar.gz
+jobs:
+  build-chart:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Install dependencies
+        run: |
+          sudo snap install yq
+          curl ${{ env.HELM_URL_V3 }} | tar xvzf - --strip-components=1 -C /tmp/ && \
+          sudo mv /tmp/helm /usr/bin/helm_v3 && \
+          sudo chmod +x /usr/bin/helm_v3
+          helm_v3 plugin install https://github.com/helm-unittest/helm-unittest.git --version ${{ env.HELM_UNITTEST_VERSION }}; \
+      - name: Build
+        run: ./scripts/chart/build chart
+      - name: Validate
+        run: ./scripts/chart/validate
+      - name: Test
+        run: ./scripts/chart/test
+      - name: Package
+        run: ./scripts/chart/package
+  build-server:
```

Comment:
It looks like we're not running any of the unit / integration tests - is that going to be added in the near future?

Reply:
Unit tests are in the `validate.yml` file, but integration tests are not ready yet, they will be done in another PR.

Reply:
From what I can see `validate.yaml` just runs linters and go mod checks. Unit tests are run in the `scripts/test` file, [here](https://github.com/rancher/rancher/blob/71c10185d52f3e1dc30b941317c71805133e6a17/scripts/test#L25-L27), just before we run the integration tests. I'm not sure we should merge this without also running the tests, but I understand getting the integration tests running in gha is no small feat.

Reply:
You're right, running `scripts/test` is being done in a different branch